### PR TITLE
e2e tests separate build job per plugin

### PR DESCRIPTION
**Pull request by sawantpritam**
@itsankit-google @rmstar Please review.

This PR contains :

1. Updated e2e tests run condition.
We should run all e2e tests on PR raise, for addition of new test scenarios or updates in existing scenario.

2. When all e2e tests are executed (on PR merge / manual trigger / any change in existing tests) - create separate parallel build jobs to execute all tests per plugin.

### CODE REVIEW

**Comment by sawantpritam** [comment]
Workflow runs:
1. Required tests
https://github.com/data-integrations/google-cloud/actions/runs/2789291303
2. All tests
https://github.com/data-integrations/google-cloud/actions/runs/2789558834

**Review by itsankit-google — COMMENTED** [review]
Can we also separate the required tests for different runners on different jobs? Please check if it can reduce some execution time.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -39,7 +38,37 @@ jobs:
...
+        id: filter
+        with:
+          working-directory: plugin
+          filters: |
+            e2e-test:
+              - '**/e2e-test/**'
+      - name: Set e2e-tests list
+        id: spl
+        run: |
+          if [ ${{ github.event_name }} != 'workflow_dispatch' ] && [ ${{ github.event_name }} != 'push' ] && [ ${{ steps.filter.outputs.e2e-test }} == 'false' ]; then
```
Comment (itsankit-google) [inline .github/workflows/e2e.yml]: suppose there is no change in e2e-tests in the PR, it will end up running every tests, right?

Comment (sawantpritam) [reply .github/workflows/e2e.yml]: No. `steps.filter.outputs.e2e-test` will be `false` when there is no change in e2e-tests so only required tests will be run.

**Comment by sawantpritam** [comment]
Rebased PR.

**Review by itsankit-google — APPROVED** [review]
Please squash commits.

**Comment by sawantpritam** [comment]
> Please squash commits.

Commits squashed.

**Pull request merged**